pkg/virt-controller/watch/vmi.go

```diff
@@ -652,6 +543,163 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
+	if !conditionsEqual(newVMI.Status.Conditions, oldVMI.Status.Conditions) {
+
+		newConditions, err := json.Marshal(newVMI.Status.Conditions)
+		if err != nil {
+			return nil, err
+		}
+		oldConditions, err := json.Marshal(oldVMI.Status.Conditions)
+		if err != nil {
+			return nil, err
+		}
+
+		patchOps = append(patchOps, fmt.Sprintf(`{ "op": "test", "path": "/status/conditions", "value": %s }`, string(oldConditions)))
+		patchOps = append(patchOps, fmt.Sprintf(`{ "op": "replace", "path": "/status/conditions", "value": %s }`, string(newConditions)))
+
+		log.Log.V(3).Object(oldVMI).Infof("Patching VMI conditions")
+	}
+
+	if !reflect.DeepEqual(newVMI.Status.ActivePods, oldVMI.Status.ActivePods) {
+		newPods, err := json.Marshal(newVMI.Status.ActivePods)
+		if err != nil {
+			return nil, err
+		}
+		oldPods, err := json.Marshal(oldVMI.Status.ActivePods)
+		if err != nil {
+			return nil, err
+		}
+
+		patchOps = append(patchOps, fmt.Sprintf(`{ "op": "test", "path": "/status/activePods", "value": %s }`, string(oldPods)))
+		patchOps = append(patchOps, fmt.Sprintf(`{ "op": "replace", "path": "/status/activePods", "value": %s }`, string(newPods)))
+
+		log.Log.V(3).Object(oldVMI).Infof("Patching VMI activePods")
+	}
+
+	if newVMI.Status.LauncherContainerImageVersion != oldVMI.Status.LauncherContainerImageVersion {
+		if oldVMI.Status.LauncherContainerImageVersion == "" {
+			patchOps = append(patchOps, fmt.Sprintf(`{ "op": "add", "path": "/status/launcherContainerImageVersion", "value": "%s" }`, newVMI.Status.LauncherContainerImageVersion))
+		} else {
+			patchOps = append(patchOps, fmt.Sprintf(`{ "op": "test", "path": "/status/launcherContainerImageVersion", "value": "%s" }`, oldVMI.Status.LauncherContainerImageVersion))
+			patchOps = append(patchOps, fmt.Sprintf(`{ "op": "replace", "path": "/status/launcherContainerImageVersion", "value": "%s" }`, newVMI.Status.LauncherContainerImageVersion))
+		}
+	}
+
+	if !reflect.DeepEqual(oldVMI.Labels, newVMI.Labels) {
+		newLabelBytes, err := json.Marshal(newVMI.Labels)
+		if err != nil {
+			return nil, err
+		}
+		oldLabelBytes, err := json.Marshal(oldVMI.Labels)
+		if err != nil {
+			return nil, err
+		}
+
+		if oldVMI.Labels == nil {
+			patchOps = append(patchOps, fmt.Sprintf(`{ "op": "add", "path": "/metadata/labels", "value": %s }`, string(newLabelBytes)))
+		} else {
+			patchOps = append(patchOps, fmt.Sprintf(`{ "op": "test", "path": "/metadata/labels", "value": %s }`, string(oldLabelBytes)))
+			patchOps = append(patchOps, fmt.Sprintf(`{ "op": "replace", "path": "/metadata/labels", "value": %s }`, string(newLabelBytes)))
+
+		}
+	}
+
+	if len(patchOps) == 0 {
+		return nil, nil
+	}
+
+	patchBuffer := bytes.Buffer{}
+	patchBuffer.WriteString("[ ")
+	for i, entry := range patchOps {
+		patchBuffer.WriteString(entry)
+		if i == len(patchOps)-1 {
+			patchBuffer.WriteString(" ]")
+		} else {
+			patchBuffer.WriteString(", ")
+		}
+	}
```

Comment:
there's a function to do this for us now `GeneratePatchBytes` in the controller pkg

Reply:
cool, changed to use it.